metrics/src/test/java/com/expedia/adaptivealerting/metrics/functions/service/MetricQueryServiceTest.java

```diff
@@ -0,0 +1,168 @@
+package com.expedia.adaptivealerting.metrics.functions.service;
+
+import com.expedia.adaptivealerting.anomdetect.util.HttpClientWrapper;
+import com.expedia.adaptivealerting.kafka.TypesafeConfigLoader;
+import com.expedia.adaptivealerting.metrics.functions.source.MetricFunctionsReader;
+import com.expedia.adaptivealerting.metrics.functions.source.MetricFunctionsSpec;
+import com.expedia.adaptivealerting.metrics.functions.source.graphite.GraphiteQueryResult;
+import com.expedia.metrics.MetricData;
+import com.expedia.metrics.MetricDefinition;
+import com.expedia.metrics.TagCollection;
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.http.entity.ContentType;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.apache.http.client.fluent.Content;
+import org.mockito.MockitoAnnotations;
+
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class MetricQueryServiceTest {
+
+    @Mock
+    private HttpClientWrapper httpClientWrapper;
+
+    private MetricData validMetricData;
+
+    private MetricData defaultMetricData;
+
+    private MetricFunctionsSpec metricFunctionsSpec;
+
+    private Config metricSourceSinkConfig;
+
+    private Config mockMetricSourceSinkConfig;
+
+    private MetricQueryService metricQueryService;
+
+    private double METRIC_DEFAULT_VALUE = 0.0;
+    private static final String METRIC_SOURCE_SINK = "metric-source-sink";
+    private static final String APP_ID = "aa-metric-functions-test";
+    private static final String NON_GRAPHITE_APP_ID = "aa-metric-functions-non-graphite-test";
+    private static final String OTHER_SOURCE_APP_ID = "aa-metric-functions-other-test";
+
+
+
+    @Before
+    public void initMetricSourceSinkConfigSetup() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        val config = new TypesafeConfigLoader(APP_ID).loadMergedConfig();
+        metricSourceSinkConfig = config.getConfig(METRIC_SOURCE_SINK);
+        val functionsInputFile = "/config/functions-test.txt";
+        metricFunctionsSpec = MetricFunctionsReader.readFromInputFile(functionsInputFile).get(0);
+        val sampleJsonGraphite = "[{" + "\"datapoints\": [[12.0, 1568255056]], " +
+                "\"target\": \"sumSeries(a.b.c)\", " +
+                "\"tags\":" + "{" + "\"aggregatedBy\": \"sum\", \"name\": \"sumSeries(a.b.c)\"" + "}}]";
```

Comment:
See comment below - consider using explicit class and serializing to avoid having to hand-craft strings.